docker-app/qfieldcloud/core/tests/test_secret_packages.py

```diff
@@ -0,0 +1,245 @@
+            "DCIM/1.jpg",
+            io.FileIO(testdata_path("DCIM/1.jpg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/2.jpg",
+            io.FileIO(testdata_path("DCIM/2.jpg"), "rb"),
+        )
+
+        # create secrets for organization and assigned to user u1
+        self._create_secret(name="SECRET", organization=self.o1, value="o1")
+        self._create_secret(
+            name="SECRET", organization=self.o1, assigned_to=self.u1, value="u1"
+        )
+
+        self.p1.refresh_from_db()
+
+        # create a package for user u1
+        repackage(self.p1, self.u1)
+
+        wait_for_project_ok_status(self.p1)
+        self.p1.refresh_from_db()
+
+        latest_package_jobs_qs = self.p1.latest_package_jobs()
+
+        self.assertEquals(latest_package_jobs_qs.count(), 1)
+
+        # create a package for user u2
+        repackage(self.p1, self.u2)
+
+        latest_package_jobs_qs = self.p1.latest_package_jobs()
+
+        self.assertEquals(latest_package_jobs_qs.count(), 2)
+
+    def test_create_secrets_and_packages(self):
+        # upload data & QGIS project files to the project.
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "bumblebees.gpkg",
+            io.FileIO(testdata_path("bumblebees.gpkg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "simple_bumblebees.qgs",
+            io.FileIO(testdata_path("simple_bumblebees.qgs"), "rb"),
+        )
+
+        # upload some attachments files to the project.
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/1.jpg",
+            io.FileIO(testdata_path("DCIM/1.jpg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/2.jpg",
+            io.FileIO(testdata_path("DCIM/2.jpg"), "rb"),
+        )
+
+        # create a project-level secret
+        self._create_secret(name="SECRET", project=self.p1, value="p1")
+
+        self.p1.refresh_from_db()
+
+        package_job_1 = repackage(self.p1, self.u1)
+        wait_for_project_ok_status(self.p1)
+        self.p1.refresh_from_db()
+
+        latest_package_ids = self.p1.latest_package_jobs().values_list("id", flat=True)
+
+        self.assertIn(package_job_1.id, latest_package_ids)
+
+        # create a user-level secret, assigned to user u1 for project p1
+        self._create_secret(
+            name="SECRET", assigned_to=self.u1, project=self.p1, value="u1"
+        )
+
+        package_job_2 = repackage(self.p1, self.u1)
+        wait_for_project_ok_status(self.p1)
+        self.p1.refresh_from_db()
+
+        latest_package_ids = self.p1.latest_package_jobs().values_list("id", flat=True)
+
+        self.assertEquals(len(latest_package_ids), 1)
+        self.assertIn(package_job_2.id, latest_package_ids)
+        self.assertNotIn(package_job_1.id, latest_package_ids)
```

Comment:
```suggestion
        latest_package_jobs_qs = self.p1.latest_package_jobs()

        self.assertEquals(latest_package_jobs_qs.count(), 1)
        self.assertIn(package_job_2, latest_package_jobs_qs.first())
```

The same spirit as the previous comments.